docs/design/proposals/carm/cross-account-resource-management.md

```diff
@@ -0,0 +1,177 @@
+>
+>Before ACK was capable of cross-account resource management, Alice would need to coordinate with the central IT team to create separate Kubernetes clusters, one tied to the AWS account her team usually used and another tied to the AWS account that RDS database resources were owned by. ACK service controllers would need to be installed into each of these separate Kubernetes clusters. Alice's team needed to have separate code repositories containing Kubernetes manifests -- one repository for the ACK custom resources owned by her team's AWS account, and another repository for the RDS database instance CRs owned by the other AWS account.
+>
+> With cross-account resource management, Alice only needs to have a single Kubernetes cluster and a single code repository to store Kubernetes configuration files for her ACK custom resources. This greatly simplifies Alice's life and reduces the amount of time she needs to coordinate with the central IT team.
+
+Today administrators can manage AWS resources using ACK. However, these resources are created in the same AWS account as the Kubernetes cluster. This leads to a suboptimal experience for entreprise customers: creating resources in multiple AWS accounts requires ACK to be installed and managed in many kubernetes clusters. Considering ACK has dozens of service controllers, this means administrators must manage hundreds of AWS service controller pods across many kubernetes clusters and many AWS Accounts with scoped roles.
+
+With Cross Account Resource Management (CARM) feature, customer(s) can install ACK on a single Kubernetes cluster which provides ability to Create/Read/Update/Delete resources in different AWS Account(s). It also allows administrators to manage permissions to access the services across accounts efficiently.
+
+## Proposed implementation
+
+#### Overview
+
+The proposed solution will solve the problem of cross account management within a single kubernetes cluster. Given an AWS Owner account ID it will look up an IAM Role ARN and call **STS::AssumeRole** to pivot the AWS SDK calls in order to execute in a new AWS Account and role context.
+
+Controllers will store AccountIDs and their associated IAM Role ARNs in a `ConfigMap` Kubernetes object. Cluster administrators will be responsible for administrating and updating the `ConfigMap`, and creating the target roles using the AWS Console/CLI. Controllers can reuse and existing IAM Role ARN if it matches the scope they need.
+
+To prevent unauthorized users from making any CRUD operations on resources in different AWS accounts, ACK will associate a unique Namespace to each AccountID, which will help administrators limit access to namespaces using K8s RBAC model, hence limit access to AccountIDs.
```

Comment:
This paragraph gets the solution a bit backwards (though the sequence diagram below gets it correct). ACK will not associate a unique Namespace to each AccountID. Rather, the cluster administrator can set an AWS Owner Account ID as an annotation on each Namespace, and when this annotation is found by the ACK service controller, the value of the annotation will override the AWS Account ID used by the ACK service controller's IAM Role.